haystack/nodes/prompt/invocation_layer/hugging_face.py

```diff
@@ -193,7 +193,8 @@ def invoke(self, *args, **kwargs):
                 model_input_kwargs["stopping_criteria"] = StoppingCriteriaList([sw])
             if top_k:
                 model_input_kwargs["num_return_sequences"] = top_k
-                model_input_kwargs["num_beams"] = top_k
+                if "num_beams" not in model_input_kwargs or model_input_kwargs["num_beams"] < top_k:
```

Comment:
For cases where `model_input_kwargs["num_beams"] < top_k`, I'd issue a warning message to let users know that we overwrite their `num_beams` parameter with `top_k`.